### PR TITLE
[WIP] Java 11 readiness: build both on JDK8 and JDK11

### PR DESCRIPTION
**Pull request by batmat**
Work in progress to make sure this plugin is constantly checked both building with JDK8 and JDK11.

This goes with the Jenkins Java 11 General Availability announcement made back in March, and we're making a pass to check plugins are looking good on a JDK11.

([to understand the Jenkinsfile content](https://wiki.jenkins.io/display/JENKINS/Java+11+Developer+Guidelines#Java11DeveloperGuidelines-MakesureyourpluginistestedinContinuousIntegrationonJava8andJava11atthesametime))

### CODE REVIEW

**Comment by batmat** [comment]
Tricky test failures AFAICT, even bumping all plugins to latest. Will need more time to fix this I think.

File: pom.xml [hunk 1]
```diff
@@ -234,6 +245,12 @@
             <scope>test</scope>
             <classifier>tests</classifier>
         </dependency>
+        <dependency>
+            <groupId>commons-net</groupId>
+            <artifactId>commons-net</artifactId>
```
Comment (imod) [inline pom.xml]: why is this now needed for tests?

Comment (batmat) [reply pom.xml]: I didn't have the time until now to look into this again, it's in the pipe, but no idea when/if I will **really** be able to get to it again. Hopefully yes :)

Comment (darxriggs) [reply pom.xml]: That's to resolve a Maven enforcer error.

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>3.18</version>
+        <version>3.42</version>
```
Comment (darxriggs) [inline pom.xml]: Now there is already 3.49.

**Comment by imod** [comment]
@darxriggs @batmat I have no idea why this stuff fails and also can't reproduce the same behaviour...

local on my mac:
- `org.jenkinsci.plugins.configfiles.buildwrapper.ConfigFileBuildWrapperWorkflowTest#withTempFilesAfterRestart` fails in when run in IDEA with java 8
- `org.jenkinsci.plugins.configfiles.buildwrapper.ConfigFileBuildWrapperWorkflowTest` all tests fail when run with maven and java 8
- when building with maven and java 11, I can't even compile and get the following error: 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project config-file-provider: Compilation failure
[ERROR] java.nio.file.NoSuchFileException: /Users/domi/work/ws/jks/config-file-provider-plugin/target/classes/META-INF/annotations/hudson.Extension
```

so in short I am not able to run anything related to this PR successfully...

**Comment by batmat** [comment]
@imod Java 11 is really supported only starting from 2.164, so I think you just missed adding `-Djenkins.version=2.164.1` to compile/build with Java 11.

Going to trigger a rebuild to see how current failures are.

**Comment by imod** [comment]
strange, I can't figure out why the test are failing...

**Comment by darxriggs** [comment]
@imod Have a look at the console output for the failing tests. You can see that a test file cannot be found. For example:

```
1.362 [p #1] [Pipeline] isUnix                                                                                                                                                                                
1.362 [p #1] [Pipeline] sh                                                                                                                                                                                    
1.614 [p #1] [p] Running shell script                                                                                                                                                                         
1.614 [p #1] + cat /tmp/j h2930637724752116221/workspace/p@tmp/config9019699264774180038tmp                                                                                                                   
1.614 [p #1] cat: /tmp/j: No such file or directory                                                                                                                                                           
1.614 [p #1] cat: h2930637724752116221/workspace/p@tmp/config9019699264774180038tmp: No such file or directory                                                                                                
```

In jenkins-test-harness v2.45 (see [changelog](https://github.com/jenkinsci/jenkins-test-harness#245-2019-jan-10)) there was this change:
> Temporary directories now by default have a space in their path, to flush out path handling mistakes. (The system property jenkins.test.noSpaceInTmpDirs=true suppresses this.)

A proper quoting of `MY_FILE` in the Jenkinsfiles used inside the tests fixes the failures.

**Comment by imod** [comment]
@darkrift oh man, I have seen the error, yes - but I would never have figured out the cause of this. Many many thanks!!!!!

**Comment by imod** [comment]
@batmat this PR is now fully reflected and working in https://github.com/jenkinsci/config-file-provider-plugin/pull/70 - going to merge it now...
